components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1529,6 +1531,23 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         // Check if snapshot file exists.
         self.ctx.snap_mgr.get_snapshot_for_applying(&key)?;
 
+        // WARNING: The checking code must be above this line.
+        // Now all checking passed.
+
+        if self.fsm.peer.local_first_replicate && !self.fsm.peer.is_initialized() {
```

Comment:
I don't think `local_first_replicate` is necessary.

Reply:
It's necessary because if `local_first_replicate` is false, it means there is no data from this peer in `pending_create_peers`.